core/modules/node/node.entity.inc

```diff
@@ -462,7 +478,15 @@ class NodeStorageController extends EntityDatabaseStorageController {
       }
       else {
         $op = 'update';
-        $return = backdrop_write_record($this->entityInfo['base table'], $entity, $this->idKey);
+        // Update the base node table if this revision is marked as active.
+        if ($entity->isActiveRevision()) {
+          $return = backdrop_write_record($this->entityInfo['base table'], $entity, $this->idKey);
+        }
+        else {
+          // @todo, should a different value be returned when saving an entity
+          // with $is_active_revision = FALSE?
+          $return = FALSE;
```

Comment:
I think this prevents contrib from being able to edit revisions? Although we don't provide an UI option, modifying a revision should be possible from the UI. I think this `else {}` statement just needs to be deleted.

Reply:
Or is the idea that if you want to edit a revision, you _must_ use `$node->saveRevision()`?

Reply:
There was discussion about whether you should be able to edit a revision at all. I think D8 makes it so if you try, it simply creates a new revision of the revision you were editing. In other words, let's decide about saving revisions or not first.